# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gperf/code_fix_gcc40.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/positions.h~	2003-04-07 11:48:19.000000000 +0200
+++ ./src/positions.h	2005-10-23 16:00:30.000000000 +0200
@@ -71,15 +71,15 @@
   bool                  sort ();
 
   /* Creates an iterator, returning the positions in descending order.  */
-  PositionIterator      iterator () const;
+  class PositionIterator      iterator () const;
   /* Creates an iterator, returning the positions in descending order,
      that apply to strings of length <= maxlen.  */
-  PositionIterator      iterator (int maxlen) const;
+  class PositionIterator      iterator (int maxlen) const;
   /* Creates an iterator, returning the positions in ascending order.  */
-  PositionReverseIterator reviterator () const;
+  class PositionReverseIterator reviterator () const;
   /* Creates an iterator, returning the positions in ascending order,
      that apply to strings of length <= maxlen.  */
-  PositionReverseIterator reviterator (int maxlen) const;
+  class PositionReverseIterator reviterator (int maxlen) const;
 
   /* Set operations.  Assumes the array is in reverse order.  */
   bool                  contains (int pos) const;
